mlir/lib/Dialect/Affine/Transforms/LoopFusion.cpp

```diff
@@ -205,7 +205,10 @@ static bool isEscapingMemref(Value memref, Block *block) {
   // (e.g., call ops, alias creating ops, etc.).
   return llvm::any_of(memref.getUsers(), [&](Operation *user) {
     // Ignore users outside of `block`.
-    if (block->getParent()->findAncestorOpInRegion(*user)->getBlock() != block)
+    auto ancestorOp = block->getParent()->findAncestorOpInRegion(*user);
```

Comment:
```suggestion
    Operation *ancestorOp = block->getParent()->findAncestorOpInRegion(*user);
```
Using `Operation` instead of `auto` because the type is not completely obvious (<https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable>).

Using the pointer declaration (`*`) is something that `clang-tidy` suggested in my editor.